src/http/server.rs

```diff
@@ -1,4 +1,37 @@
 //! HTTP server
+//!
+//! Provides an HTTP(S) server in `EspHttpServer`, plus all related structs.
+//!
+//! Typical usage of `EspHttpServer` involves creating a function (or closure)
+//! for every URI+method that the server is meant to handle. A minimal server that
+//! only handles HTTP GET requests to `index.html` looks like this:
+//!
+//! ```
+//! use esp_idf_svc::http::server::{Configuration, EspHttpServer};
+//!
+//! let mut server = EspHttpServer::new(&Configuration::default())?;
+//!
+//! server.fn_handler("/index.html", Method::Get, |request| {
+//! 	request.into_ok_response()?
+//! 		.write_all("<html><body>Hello world!</body></html>".as_bytes());
+//!
+//! 	Ok(());
+//! })?;
+//! ```
+//!
+//! Note that the server is automatically started when instantiated, and stopped
```

Comment:
Maybe nitpicking again, but 
* remove "Due to this, a common practice is to run the HTTP server in its own thread.". This is not a common practice, and I don't see the point of doing this. You just need to have the server instance around, that's all
* "and when its references are dropped" -> "and stopped when dropped"

... or I can do that too post-commit.

Reply:
Changed to "If you want to keep the server running indefinitely then make sure it's not dropped - you may add an infinite loop after the server is created, use core::mem::forget, or keep around a reference to it somehow.".

(Doesn't show as a change since it's a couple of lines down)